### PR TITLE
sem86/add copy button to code snippets

### PR DESCRIPTION
**Pull request by sophiemachin**
Add javsacript and styling for a copy button for code snippets

### CODE REVIEW

**Review by rsa33 — CHANGES_REQUESTED** [review]
One initial change request to get this to work across all of the docs...

**Comment by rsa33** [comment]
Some further considerations (not a direct problem with this PR, but things we probably want to fix if we bring this in):

* We have some non-highlighted code blocks which could also benefit from Copy buttons (e.g. [Setting up IRC](https://docs.srcf.net/tutorials/others/connect-to-irc/)).  We could target `<pre>` tags here instead with `document.getElementsByTagName("pre")`, though this may have some unintended side-effects with placing a button inside a preformatted block; it will also need `position: relative` applying to `pre`.  Alternatively we can just remove the (invalid?) highlight types, or fix the theme to always wrap code blocks in a `<div class="highlight">`.

* We also have a mix of shell prefixes (e.g. `$` for bash samples) vs. no prefixes in various code samples, which will also get copied.  We'll want to remove them if we bring in block copying, but it would be nice to still render them somehow (possibly with CSS `:before` + `content` magic to add a visual-only prefix).

* Most of our code blocks have an unnecessary 4-space indent, which gets included in the copied text.  We should dedent those in the page content (and add horizontal padding if necessary).

File: layouts/index.html [hunk 1]
```diff
@@ -124,5 +124,6 @@ <h2>{{ $heading }}</h2>
         >
       </div>
     </div>
+    <script src="/js/copy-button.js"></script>
```
[the remark this answers is not in the document]
Comment (edwinbalani) [reply layouts/index.html:127]: I'm trying to make it work in there, but it's just not appearing when I do a `make clean build`.  I see the existing bootstrap `<script>` tag, but adding anything (even 'foo') to the partial doesn't change where I think it should be output.  I'm a bit stumped right now.

[the remark this answers is not in the document]
Comment (edwinbalani) [reply layouts/index.html:127]: ...found the issue.  I was missing a `</script>`.  Looks like Hugo filtered the whole tag out because of that.

**Comment by CMTC** [comment]
I'm not sure this change makes sense; I think the pattern of copying and execute anything one sees online without thinking about it is a harmful one. In any case, I think a lot of code blocks are actually used in the documentation to show examples of command prompts and such (including some hypothetical value of $PS1), so copying the whole thing is likely to be inappropriate.

In any case, is copying so difficult anyway? On my machines, I normally just sweep text with the mouse and paste with middle click (X11 primary selection).

**Comment by edwinbalani** [comment]
> In any case, is copying so difficult anyway? On my machines, I normally just sweep text with the mouse and paste with middle click (X11 primary selection).

I think you're in the intersectional minority of people who are using X11, and know what the primary selection is, and are always using a 3-button mouse.  We won't expect that of everyone.

**Comment by matiasilva** [comment]
I just wanted to chime in and say thank you for your pull request! We really want to foster a culture where community contributions are welcome and respected. On a more personal note, as someone who redesigned the current docs, I'm glad to see people adding on useful functionality to it. Thanks again :)

**Review by rsa33 — APPROVED** [review]
Thanks for that -- looks good to me.  👍

I think this is a useful addition for the majority of visitors, and will behave gracefully in the presence of no JavaScript (by just not adding the buttons in the first place).

**Comment by sophiemachin** [comment]
Is there anything more to do with this? I don't have permissions to merge.

**Comment by rsa33** [comment]
Sorry for the delay -- there were some minor reservations elsewhere blocking the linked code snippet changes, but overall the combined changes are net-positive improvement, so let's get this live.  🙂

**Pull request merged**